### PR TITLE
Make `version` subcommand highlight outdated apps, refs #104

### PR DESCRIPTION
**Pull request by clochix**

### CODE REVIEW

File: lib/application.coffee [hunk 1]
```diff
@@ -374,6 +374,11 @@ module.exports.getVersion = (app, callback) ->
     callback app.version
 
 
+# Callback application needs update
+module.exports.needsUpdate = (app, callback) ->
+    callback app.needsUpdate
```
Comment (frankrousseau) [inline lib/application.coffee]: Why do use a callback here?

Comment (clochix) [inline lib/application.coffee]: Why do we need a callback in the previous `getVersion` method that just returns an attribute of the app object? I followed the coding style, be agree we don't need it.

**Comment by aenario** [comment]
Should we merge this ? Ping @nledez and @LucsT, could this break some of our monitoring scripting ?

**Comment by aenario** [comment]
After discussion with @nledez, we should NOT MERGE THIS PR as-is. 

The path forward would be to :
1 - add a raw option to cozy monitor commands with JSON output 
2 - adapt infra scripting to use said JSON output instead of parsing cozy-monitor text
3 - Merge this PR and add any further goodies to the human readable without fear.

**Comment by nledez** [comment]
Or give me an other parameter such as "versions.json". :)

**Comment by clochix** [comment]
I just added a `--json` option to `versions` subcommand.
@nledez: which other command would you like to have such an option?

File: lib/application.coffee (hunk 1 above)
Comment (frankrousseau) [inline lib/application.coffee]: Ok can you remove it?

**Comment by clochix** [comment]
@nledez I added a `--json` option to the `status` subcommand (but need to fix the tests). Anything else?

**Comment by frankrousseau** [comment]
Linting is wrong. Can you pass a linter (space missing after comma in an array, space missing after `:` in an object...) ?

**Comment by nledez** [comment]
If I can made my list to santa:
- cozy-monitor versions
- cozy-monitor status

And an extra list:
- cozy-monitor module-status <module>
- cozy-monitor views-list

**Comment by clochix** [comment]
@frankrousseau which linter are you using? Coffeelint is all green with the default parameters.

**Comment by frankrousseau** [comment]
:_( I think we need to change the default parameters then. It's sad it doesn't warn you about the blank space mistakes. There is this one for commas `spacing_after_comma`. But I don't see anything for `:`.

**Comment by clochix** [comment]
Don't worry, when we'll have completed the migration to ES-2015, we'll be able to use lot of linters, like jshint, eslint, jscs ;-)

**Comment by frankrousseau** [comment]
Not sure, when you will read many new blog posts about why Typescript is the future of javascript you will say that ES6 is so 2015 that you don't want to think about it anymore! ;)

**Comment by clochix** [comment]
I added a json option for module-status and views-list, and now tests are green.

**Comment by nledez** [comment]
:thumbsup:

**Comment by aenario** [comment]
@nledez do you mean :+1: this is cool, I cant wait to change the scripts using this or :+1: I have changed the scripts using this and a passing TR can merge this PR ?

**Comment by frankrousseau** [comment]
@aenario let's merge it!

**Comment by nledez** [comment]
@aenario this is cool! :thumbsup: And in can deal with this change :) Thanks!

**Pull request merged**